### DISCUSSION TITLE
RC6.50 Beta

### DISCUSSION THREAD

**Discussion by electricrt** [electricrt/nxt_firmware_updates#2] [Announcements]
**NXT 6.50 Beta**

**General Fixes**

Patched possible save bug *done*

Can now run the BAC headless without the NXT plugged in, where as before the BAC would not run without the NXT plugged in. *done*

Patched under and overvoltage cut offs for 78V / 21S+ batteries

Real time clock added

Adjustable brightness

**Input settings**

Throttle calibration now works with Potentiometer throttles. Previously this was bugged as the min throttle value was 0.8v.

**Invert Brakes**

Added option to invert brakes, to account for normally open and normally closed closed brakes. 

**Main Screen**

Fixed Voltage Reading, now shows voltage up to 2 decimals and the V label

**Odometer / Tripmeter**

Odometer and TripMeter now correctly saves to memory

**Stats** 

Fixed nominal voltage (before it was always just showing 52V) *done*

**Diagnostics** 

Revamped Diagnostics, if there is a fault or warning detected. Will try to auto clear fault first.  *done*

There will be a notification on the mainscreen that indicates that. Pressing the notification will open the diagnostics page where you can see the exact fault or warning. 

**Auto fault clearing** *done*

If there is a fault, it will attempt to automatically clear itself. If unsuccessful then proceed to diagnostics page. 

**Motor Setup** *done*

Completely overhauled.
Also added invert motor direction 

**Sur Ron**

**CYC X1 Pro**

**Bafang BBSHD**

Throttle Power Limit increase from 3250W to 4000W *done*

Restore will increase max motor current from 90A to 108A

Field Weakening Limit increased from 40% to 50% *done*

**LR Big Block XL**

Throttle Power Limit increased from 9000W to 15000W *done*

**New Settings**

Throttle Ramp Slider

Current Ramp Slider 

Motor IDLE Speed Slider

Assist Ramp Slider

Assist Max Speed Slider

Speed Sensor Selection (Use Motor for speedometer signal or use external speed sensor)

Adjustable battery foldback

**Comment by papagino007** [comment]
Some issues with this update...
Posted on the Issues board here: https://github.com/electricrt/nxt_firmware_updates/issues/12
